tests/basic/mutex.rs

```diff
@@ -114,6 +116,125 @@ fn concurrent_increment() {
     });
 }
 
+#[test]
+/// Two concurrent threads trying to do an atomic increment using `try_lock`.
+/// One `try_lock` must succeed, while the other may or may not succeed.
+/// Thus we expect to see final values 1 and 2.
+fn concurrent_try_increment() {
+    let final_value_tracker = Arc::new(
+        (1usize..=2usize)
+            .map(|i| (i, std::sync::atomic::AtomicBool::new(false)))
+            .collect::<HashMap<_, _>>(),
+    );
+    let final_value_tracker_clone = Arc::clone(&final_value_tracker);
+
+    check_dfs(
+        move || {
+            let lock = Arc::new(Mutex::new(0usize));
+
+            let threads = (0..2)
+                .map(|_| {
+                    let lock = Arc::clone(&lock);
+                    thread::spawn(move || match lock.try_lock() {
+                        Ok(mut guard) => {
+                            *guard += 1;
+                        }
+                        Err(TryLockError::WouldBlock) => (),
+                        Err(_) => panic!("unexpected TryLockError"),
+                    })
```

Comment:
I mildly dislike what rustfmt does here; adding a semicolon at the end of the closure body makes it more palatable to me:
```suggestion
                    thread::spawn(move || {
                        match lock.try_lock() {
                            Ok(mut guard) => {
                                *guard += 1;
                            }
                            Err(TryLockError::WouldBlock) => (),
                            Err(_) => panic!("unexpected TryLockError"),
                        };
                    })
```